### ISSUE TITLE
Port twisted.python.lockfile to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5960]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5960|
|Type|enhancement|
|Created|2012-09-11 14:55:18Z|
|Branch|https://github.com/twisted/twisted/tree/lockfile-py3-5960|

lockfile is required by trial (and technically `twisted.internet.defer`). We could probably work around it but it's trivial to port.

Attachments:

* [lockfile-r0.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5f6/5f6fcf64d1621b400a123fbc8ad45083f0deb128/484466b22963f7d4a1987b5ba68dcdcc2f843899.patch) (3141 bytes) - added by jamesbroadhead on 2013-07-06 13:41:41Z - 
* [lockfile-r1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5f6/5f6fcf64d1621b400a123fbc8ad45083f0deb128/bd5e767da57617cab07e92231b326db62c847a66.patch) (19925 bytes) - added by jamesbroadhead on 2013-07-06 15:37:01Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5960 5960
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_lockfile_py3_5960 branches/lockfile-py3-5960
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1347375318000000 1347375318000000
changetime__1373451835000000 1373451835000000
version__None None
owner__therve therve

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

This should also add Python 3 support for `twisted.internet.defer.DeferredFilesystemLock`.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve commented|
|-|-|

(In [39100]) Branching to 'lockfile-py3-5960'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set owner to @therve|
|-|-|

Looks great, applying!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve set status to `closed`|
|-|-|

(In [39103]) Merge lockfile-py3-5960

Author: jamesbroadhead
Reviewer: therve
Fixes: [#5960](https://github.com/twisted/twisted/issues/5960)

Port twisted.python.lockfile to python3.
